submissions/andrewklmn/a-tiny-js-world/index.js

```diff
@@ -6,56 +6,117 @@
+  if (this.friends.length > 0) {
+    return this.friends.map(friend => friend.name).join(', ');
+  };
+  return 'No friends!';
+};
+Inhabitant.prototype.toString = function () {
+  return `${this.species}; ${this.name}; ${this.gender}; ${this.legs}; ${this.hands}; ${this.saying}; ${this.getFriendsList()}`;
 };
 
-const cat = {
-  species: 'cat',
-  name: 'Barsik',
-  gender: 'male',
-  legs: 4,
-  hands: 0,
-  saying: 'meow!',
-  friends: [],
+const Pet = function ({ species, name, gender, saying }) {
+  const legs = 4;
+  Inhabitant.call(this, { species, name, gender, legs, saying });
 };
+Pet.prototype = Object.create( Inhabitant.prototype);
+Pet.prototype.constructor = Inhabitant;
 
-const woman = {
-  species: 'human',
-  name: 'Leeloo Dallas',
-  gender: 'female',
-  legs: 2,
-  hands: 2,
-  saying: 'People hi!',
-  friends: [],
+const Dog = function ({ name, gender }) {
+  const species = 'dog';
+  const saying = 'woof!';
+  Pet.call(this, {species, name, gender, saying});
 };
+Dog.prototype = Object.create(Pet.prototype);
+Dog.prototype.constructor = Pet;
 
-const man = {
-  species: 'human',
-  name: 'Korben Dallas',
-  gender: 'male',
-  legs: 2,
-  hands: 2,
-  saying: 'Hello there!',
-  friends: [],
+const MaleDog = function (name) {
+  const gender = 'male';
+  Dog.call(this, { name, gender });
 };
+MaleDog.prototype = Object.create(Dog.prototype);
+MaleDog.prototype.constructor = Dog;
 
-// define cat-woman
-const catWoman = { ...woman };
-catWoman.name = 'Cat-woman';
-catWoman.saying = cat.saying;
+const FemaleDog = function (name) {
+  const gender = 'female';
+  Dog.call(this, { name, gender });
+};
+FemaleDog.prototype = Object.create(Dog.prototype);
+FemaleDog.prototype.constructor = Dog;
+
+const Cat = function ({ name, gender }) {
+  const species = 'cat';
+  const saying = 'meow!';
+  Pet.call(this, {species, name, gender, saying });
+};
+Cat.prototype = Object.create(Pet.prototype);
+Cat.prototype.constructor = Pet;
+
+const MaleCat = function (name) {
+  const gender = 'male';
+  Cat.call(this, { name, gender });
+};
+MaleCat.prototype = Object.create(Cat.prototype);
+MaleCat.prototype.constructor = Cat;
+
+const FemaleCat = function (name) {
+  const gender = 'female';
+  Cat.call(this, { name, gender });
+};
+FemaleCat.prototype = Object.create(Cat.prototype);
+FemaleCat.prototype.constructor = Cat;
+
+const Human = function ({ name, gender, saying}) {
+  const species = 'human';
+  const legs = 2;
+  const hands = 2;
+  Inhabitant.call(this, { species, name, gender, legs, hands, saying });
+};
+Human.prototype = Object.create(Inhabitant.prototype);
+Human.prototype.constructor = Inhabitant;
+
+const Woman = function ({ name, saying }) {
+  const gender = 'woman';
+  Human.call(this, { name, gender, saying });
+};
+Woman.prototype = Object.create(Human.prototype);
+Woman.prototype.constructor = Human;
+
+const Man = function ({ name, saying }) {
+  const gender = 'man';
+  Human.call(this, { name, gender, saying });
```

Comment:
```suggestion
  Human.call(this, { name, gender: 'man', saying });
```
Just as a proposition :)